src/utils/runtime/src/lib.rs

```diff
@@ -228,13 +231,75 @@ pub fn enable_parking_lot_deadlock_detection() {
     });
 }
 
-/// Common set-up for all RisingWave binaries. Currently, this includes:
+fn spawn_prof_thread(profile_path: String) -> std::thread::JoinHandle<()> {
+    tracing::info!("writing prof data to directory {}", profile_path);
+    let profile_path = PathBuf::from(profile_path);
+
+    std::fs::create_dir_all(&profile_path).unwrap();
+
+    std::thread::spawn(move || {
+        let mut cnt = 0;
+        loop {
+            let guard = pprof::ProfilerGuardBuilder::default()
+                .blocklist(&["libc", "libgcc", "pthread", "vdso"])
+                .build()
+                .unwrap();
+            std::thread::sleep(Duration::from_secs(60));
+            match guard.report().build() {
+                Ok(report) => {
+                    let profile_svg = profile_path.join(format!("{}.svg", cnt));
+                    let file = std::fs::File::create(&profile_svg).unwrap();
+                    report.flamegraph(file).unwrap();
+                    tracing::info!("produced {:?}", profile_svg);
+                }
+                Err(err) => {
+                    tracing::warn!("failed to generate flamegraph: {}", err);
+                }
+            }
+            cnt += 1;
+        }
+    })
+}
+
+/// Start RisingWave components with configs from environment variable.
 ///
-/// * Set panic hook to abort the whole process.
-pub fn oneshot_common() {
+/// Currently, the following env variables will be read:
+///
+/// * `RW_WORKER_THREADS`: number of tokio worker threads. If not set, it will use tokio's default
+///   config (equivalent to CPU cores).
+/// * `RW_DEADLOCK_DETECTION`: whether to enable deadlock detection. If not set, will enable in
+///   debug mode, and disable in release mode.
+/// * `RW_PROFILE_PATH`: the path to generate flamegraph. If set, then profiling is automatically
+///   enabled.
```

Comment:
How we expect devs to know this? Only be looking at the code doc? 🤔 
Will we provide a config file?

Reply:
I think this is only helpful for developers, so it won't be added in user docs. For developers, currently I have no idea how to gain more visibility on these parameters. Let's keep it in code doc for now.